M2/Macaulay2/d/evaluate.d

```diff
@@ -1268,46 +1268,45 @@ augmentedAssignmentFun(x:augmentedAssignmentCode):Expr := (
 	    is Nothing do nothing
 	    else return e)
 	else lexpr = eval(x.lhs);
-	left := evaluatedCode(lexpr, dummyPosition);
-	when left.expr is e:Error do return Expr(e) else nothing;
+	when lexpr is e:Error do return lexpr else nothing;
 	-- check if user-defined method exists
-	meth := lookup(Class(left.expr),
-	    Expr(SymbolClosure(globalFrame, x.oper)));
+	meth := lookup(Class(lexpr), Expr(SymbolClosure(globalFrame, x.oper)));
 	if meth != nullE then (
-	    rightexpr := eval(x.rhs);
-	    when rightexpr is e:Error do return(e) else nothing;
-	    r := applyEEE(meth, left.expr, rightexpr);
+	    rexpr := eval(x.rhs);
+	    when rexpr is e:Error do return rexpr else nothing;
+	    r := applyEEE(meth, lexpr, rexpr);
 	    when r
 	    is s:SymbolClosure do (
 		if s.symbol.word.name === "Default" then nothing
 		else return r)
 	    else return r);
 	-- if not, use default behavior
+	left := evaluatedCode(lexpr, codePosition(x.lhs));
```

Comment:
I know writing tests for errors is complicated, but did you try to reproduce an error at the errors points below?

Reply:
Oh yeah -- they're easy to reproduce.  For example:

```m2
i1 : x = 5

o1 = 5

i2 : x += "foo"
stdio:2:2:(3): error: no method for binary operator + applied to objects:
            5 (of class ZZ)
      +     "foo" (of class String)
```